File: 백준/Bronze/6975. Deficient， Perfect， and Abundant/Deficient， Perfect， and Abundant.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    N = int(input())
    for n in range(N):
        if n > 0:
            print()
        X = int(input())
        lst = []
        for i in range(1, X//2+1):
            if X%i == 0:
                lst.append(i)
        Y = sum(lst)
        if Y > X:
            print(f'{X} is an abundant number.')
        elif Y == X:
            print(f'{X} is a perfect number.')
        else:
            print(f'{X} is a deficient number.')


if __name__ == '__main__':
    main()

File: 백준/Bronze/6975. Deficient， Perfect， and Abundant/Deficient， Perfect， and Abundant.c
#include <stdio.h>

int F(int X) {
    int sum = 0;
    for (int i = 1; i <= X / 2; i++) {
        if (X % i == 0) {
            sum += i;
        }
    }
    return sum;
}

int main() {
    int N;
    scanf("%d", &N); 
    for (int n = 0; n < N; n++) {
        if (n > 0) {
            printf("\n"); 
        }
        int X;
        scanf("%d", &X);
        int sum = F(X);
        if (sum > X) {
            printf("%d is an abundant number.\n", X);
        } else if (sum == X) {
            printf("%d is a perfect number.\n", X);
        } else {
            printf("%d is a deficient number.\n", X);
        }
    }
    return 0;
}